server/protocols/webadmin/webadmin-data/src/test/java/org/apache/james/webadmin/routes/DLPRulesRoutesTest.java

```diff
@@ -0,0 +1,410 @@
+        webAdminServer = WebAdminUtils.createWebAdminServer(
+            new DefaultMetricFactory(),
+            new DLPRulesRoutes(dlpConfigurationStore, domainList, new JsonTransformer()));
+        webAdminServer.configure(NO_CONFIGURATION);
+        webAdminServer.await();
+
+        requestSpecification = buildRequestSpecification(webAdminServer);
+    }
+
+    RequestSpecification buildRequestSpecification(WebAdminServer server) {
+        return new RequestSpecBuilder()
+            .setContentType(ContentType.JSON)
+            .setAccept(ContentType.JSON)
+            .setBasePath(DLPRulesRoutes.BASE_PATH)
+            .setPort(server.getPort().getValue())
+            .setConfig(newConfig().encoderConfig(encoderConfig().defaultContentCharset(StandardCharsets.UTF_8)))
+            .build();
+    }
+
+
+    @BeforeEach
+    void setup() throws Exception {
+        DNSService dnsService = mock(DNSService.class);
+        Mockito.when(dnsService.getHostName(any())).thenReturn("localhost");
+        Mockito.when(dnsService.getLocalHost()).thenReturn(InetAddress.getByName("localhost"));
+
+        MemoryDomainList domainList = new MemoryDomainList(dnsService);
+        domainList.setAutoDetectIP(false);
+        domainList.addDomain(SENDER_DOMAIN);
+
+        dlpStore = new EventSourcingDLPConfigurationStore(new InMemoryEventStore());
+        createServer(dlpStore, domainList);
+    }
+
+
+    @Nested
+    class DefineStore {
+        @Test
+        void putShouldstoreTheConfigurations() {
+            String body = JsonNodeFactory.instance.arrayNode()
+                .add(JSON_WITH_POSITIVE_TARGETS)
+                .add(JSON_WITH_NEGATIVE_TARGETS)
+                .toString();
+
+            given()
+                .body(body)
+            .when()
+                .put(DEFAULT_DOMAIN)
+            .then()
+            .log().all()
+                .statusCode(HttpStatus.NO_CONTENT_204);
+
+            assertThat(dlpStore.list(SENDER_DOMAIN))
+                .hasSize(2)
+                .extracting(dlpConfiguration -> dlpConfiguration.getId().asString())
+                .containsExactlyInAnyOrder("1", "2");
+        }
+
+        @Test
+        void putShouldstoreTheConfigurationsWhenTargetsAreNotSpecified() {
+            String body = JsonNodeFactory.instance.arrayNode()
+                .add(JSON_WITHOUT_TARGET)
+                .toString();
+
+            given()
+                .body(body)
+            .when()
+                .put(DEFAULT_DOMAIN)
+            .then()
+                .statusCode(HttpStatus.NO_CONTENT_204);
+
+            assertThat(dlpStore.list(SENDER_DOMAIN))
+                .hasSize(1)
+                .extracting(dlpConfiguration -> dlpConfiguration.getId().asString())
+                .containsOnly("3");
+        }
+
+        @Test
+        void putShouldReturnBadRequestWhenIdIsNotSpecified() {
```

Comment:
That's a shame to let ID generation to the ser. IMO it should be the other way around: we reject ID's set by the user, and generate it in @PUT